File: EXAM_PREPARATION_I/Skeletons/JavaScript Skeleton/models/Film.js
const mongoose = require('mongoose');

let filmSchema = mongoose.Schema({
    //TODO: Implement me ...
    /*
    •	id – technology-dependent identifier (ObjectID for JavaScript, int for all other technologies)
    •	name – non-empty text
    •	genre – non-empty text
    •	director – non-empty text
    •	year – non-null integer
     */
    name: {type: "string", required: "true"},
    genre: {type: "string", required: "true"},
    director: {type: "string", required: "true"},
    year: {type: "number", required: "true"}
});

let Film = mongoose.model('Film', filmSchema);

module.exports = Film;

File: EXAM_PREPARATION_I/Skeletons/Java Skeleton/src/main/java/imdb/entity/Film.java
package imdb.entity;

import javax.persistence.*;

@Entity
@Table(name = "films")
public class Film {
	//TODO: Implement me ...
    /*
    •	id – technology-dependent identifier (ObjectID for JavaScript, int for all other technologies)
    •	name – non-empty text
    •	genre – non-empty text
    •	director – non-empty text
    •	year – non-null integer
     */

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    @Column(nullable = false)
    private String name;

    @Column(nullable = false)
    private String genre;

    @Column(nullable = false)
    private String director;

    @Column(nullable = false)
    private Integer year;

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getGenre() {
        return genre;
    }

    public void setGenre(String genre) {
        this.genre = genre;
    }

    public String getDirector() {
        return director;
    }

    public void setDirector(String director) {
        this.director = director;
    }

    public Integer getYear() {
        return year;
    }

    public void setYear(Integer year) {
        this.year = year;
    }
}
